### ISSUE TITLE
[Development] COGNITION: Form review card edit can not be cancelled

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/va.gov-team#4877]
## Description

This issue was originally brought up with a Higher-Level Review a11y review ([ref](https://github.com/department-of-veterans-affairs/va.gov-team/issues/3566#issuecomment-559214317)), but is an issue associated with all forms. cc @jenstrickland

The problem is with the ReviewCardField component (at least two copies exist in the eBenefits & 526 all-claims forms) possibly _trapping_ the user on the page. When the user accidentally deletes or modifies the content of a review card, there is no method to cancel the changes. This may happen if the user forgets the previous content or the user is unaware of how to undo the changes (<kbd>⌘</kbd>+<kbd>Z</kbd>) or is forced to reload the page to restore the value. Note: moving back one page and continuing again will not restore the value.

<img src="https://user-images.githubusercontent.com/57469/69751235-abf5f180-111c-11ea-8867-617f6b5919f3.png" alt="phone & email review card in edit mode" />

## Acceptance Criteria

- [ ] The review card includes a "Cancel" button
- [ ] The "Cancel" button must cancel any changes (restore previous state) made within a review card when used
- [ ] The cancel button will meet all a11y criteria

## Steps to Recreate

1. Log in as user 218, then go to a form that contains a phone & email or mailing address review card, e.g. https://staging.va.gov/disability/file-disability-claim-form-21-526ez/contact-information
2. Click the "Edit" button.
3. Press <kbd>Tab</kbd> to move focus to the first input.
4. All content _should_ be selected, if not manually select it.
5. Delete the content
6. At this point,
    * pressing "Done" or "Continue" will not allow the user to progress & will not restore the value
    * pressing "Back" and "Continue" again will not restore the value
    * using "Finish this application later" will not restore the value
    * reloading the page will make you go through the ITF screen before returning to the current page, and even that will not restore the value.
    * Only immediately using undo (<kbd>⌘</kbd>+<kbd>Z</kbd>) will restore the value.

## Possible Fixes (optional)

Add a cancel button next to the "Done" button to allow the user to reset to their previous state.

![Screen Shot 2020-01-15 at 9 43 50 AM](https://user-images.githubusercontent.com/136959/72447824-a2335680-377b-11ea-859f-dac84680b576.png)

**Comment by jenstrickland** [comment]
cc @1Copenut

**Comment by 1Copenut** [comment]
@jenstrickland Agreed. We should have an `Undo` or `Cancel` that cancels the current action and restores the previous known good value. I added the "sitewide" label so this will show up in the product accessibility page's [sitewide saved search](https://github.com/department-of-veterans-affairs/va.gov-team/issues?q=is%3Aopen+is%3Aissue+label%3Asitewide+label%3A508%2FAccessibility).

**Comment by Mottie** [comment]
Created this PR: https://github.com/department-of-veterans-affairs/vets-website/pull/12013
